src/MySqlConnector/MySqlClient/Caches/CachedParameter.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Data;
+
+namespace MySql.Data.MySqlClient.Caches
+{
+	internal class CachedParameter
+	{
+		public CachedParameter(int ordinalPosition, string mode, string name, string dataType, bool unsigned)
+		{
+			Position = ordinalPosition;
+			if (Position == 0)
+			{
+				Direction = ParameterDirection.ReturnValue;
+			}
+			else
+			{
+				switch (mode.ToLower())
+				{
+					case "IN":
+						Direction = ParameterDirection.Input;
+						break;
+					case "INOUT":
+						Direction = ParameterDirection.InputOutput;
+						break;
+					case "OUT":
+						Direction = ParameterDirection.Output;
+						break;
+				}
+			}
+			Name = name;
+			switch (dataType.ToLower())
```

Comment:
Wondering if we need to implement our own `MySqlType` (or similar) that's the superset of MySQL Server data type names, .NET types, `DbType` enumeration values, `ColumnType` enumeration values, etc., etc.